core/src/com/unciv/logic/map/tile/TileInfoImprovementFunctions.kt

```diff
@@ -110,13 +110,14 @@ class TileInfoImprovementFunctions(val tile: Tile) {
 
             // Otherwise, we can if this improvement removes the top terrain
             if (!hasUnique(UniqueType.RemovesFeaturesIfBuilt, stateForConditionals)) return false
-            if (knownFeatureRemovals == null) return false
-            val featureRemovals = tile.terrainFeatures.map {
-                feature -> tile.ruleset.tileRemovals.firstOrNull{ it.name == Constants.remove + feature } }
-            if (featureRemovals.any { it != null && it !in knownFeatureRemovals }) return false
+            if (knownFeatureRemovals.isNullOrEmpty()) return false
+            val featureRemovals = tile.terrainFeatures.mapNotNull { feature -> 
+                tile.ruleset.tileRemovals.firstOrNull { it.name == Constants.remove + feature } }
+            if (featureRemovals.isEmpty()) return false
+            if (featureRemovals.any { it !in knownFeatureRemovals }) return false
```

Comment:
What does it mean, if a feature removal is not "known"? It's in the ruleset, what are we actually checking here?

Reply:
Which removals have we unlocked the techs for. Definition is further up; line 82

Reply:
The check here is "of all of the removals that could apply, do we have all of their techs unlocked?" I would go "huh, wait, why don't we check that directly?" but it was like that before. I would assume there's a reason... Hopefully

Reply:
Looks like this whole thing requires cleanup, not for this pr